<?xml version="1.0"?>
<launch>
    <include file="$(find amigo_machine_files)/machines/amigo1.machine" />
    <include file="$(find amigo_machine_files)/machines/amigo2.machine" />
    <include file="$(find amigo_machine_files)/machines/amigo3.machine" />
    <include file="$(find amigo_machine_files)/machines/amigo4.machine" />
    <include file="$(find amigo_machine_files)/machines/localhost.machine" />

    <arg name="machine" default="localhost"/>

    <node pkg="arm_kinematics_constraint_aware" type="arm_kinematics_constraint_aware" name="amigo_left_arm_kinematics"  output="screen" machine="$(arg machine)">
        <param name="group" type="string" value="left_arm" />
        <param name="left_arm/root_name" type="string" value="shoulder_mount_left" />
        <param name="left_arm/tip_name" type="string" value="grippoint_left" />
        <param name="kinematics_solver" type="string" value="arm_kinematics_constraint_aware/KDLArmKinematicsPlugin" />
        <param name="left_arm/max_search_iterations" value="1" />
        <param name="left_arm/max_solver_iterations" value="1" />
    </node>
    <node pkg="arm_kinematics_constraint_aware" type="arm_kinematics_constraint_aware" name="amigo_right_arm_kinematics"  output="screen" machine="$(arg machine)">
        <param name="group" type="string" value="right_arm" />
        <param name="right_arm/root_name" type="string" value="shoulder_mount_right" />
        <param name="right_arm/tip_name" type="string" value="grippoint_right" />
        <param name="kinematics_solver" type="string" value="arm_kinematics_constraint_aware/KDLArmKinematicsPlugin" />
        <param name="right_arm/max_search_iterations" value="1" />
        <param name="right_arm/max_solver_iterations" value="1" />
    </node>
</launch>
